conans/client/installer.py

```diff
@@ -107,7 +107,12 @@ def _prepare_sources(self, conanfile, pref, package_layout, conanfile_path, sour
     def _build(self, conanfile, pref, build_folder):
         # Read generators from conanfile and generate the needed files
         logger.info("GENERATORS: Writing generators")
-        write_generators(conanfile, build_folder, self._output)
+
+        install_folder = build_folder
+        if conanfile.lyt:
+            install_folder = os.path.join(build_folder, conanfile.lyt.build_install_folder)
```

Comment:
I would be really happy for layouts if they allow me to override the install_folder! See #6691